client/landing/stepper/declarative-flow/internals/steps-repository/sensei-plan/index.tsx

```diff
@@ -0,0 +1,396 @@
+		if ( goToStep ) {
+			goToStep( 'senseiDomain' );
+		}
+	}, [ goToStep ] );
+
+	const onPlanSelect = async () => {
+		try {
+			setStatus( Status.Bundling );
+
+			// Wait for a bit to get an animation in the beginning of site creation.
+			await new Promise( ( res ) => setTimeout( res, 100 ) );
+
+			setProgress( {
+				percentage: 25,
+				title: siteProgressTitle,
+			} );
+
+			await createSenseiSite( {
+				username: currentUser?.username || '',
+				languageSlug: locale,
+				visibility,
+			} );
+
+			setProgress( {
+				percentage: 50,
+				title: cartProgressTitle,
+			} );
+
+			const newSite = getNewSite();
+			const siteId = newSite?.blogid;
+			setSelectedSite( newSite?.blogid );
+			await setIntentOnSite( newSite?.site_slug as string, SENSEI_FLOW );
+			await saveSiteSettings( newSite?.blogid as number, { launchpad_screen: 'full' } );
+
+			setProgress( {
+				percentage: 75,
+				title: styleProgressTitle,
+			} );
+
+			if ( siteId ) {
+				const selectedStyleVariationTitle = getSelectedStyleVariation()?.title;
+				const styleVariations: StyleVariation[] = await getStyleVariations( siteId, 'pub/course' );
+				const theme: Theme = await getSiteTheme( siteId, 'pub/course' );
+				const userGlobalStylesLink: string =
+					theme?._links?.[ 'wp:user-global-styles' ]?.[ 0 ]?.href || '';
+				const userGlobalStylesId = parseInt( userGlobalStylesLink.split( '/' ).pop() || '', 10 );
+				const styleVariation = styleVariations.find(
+					( variation ) => variation.title === selectedStyleVariationTitle
+				);
+				if ( styleVariation && userGlobalStylesId && ! isNaN( userGlobalStylesId ) ) {
+					await updateGlobalStyles( siteId, userGlobalStylesId, styleVariation );
+				}
+			}
+			setProgress( {
+				percentage: 100,
+				title: styleProgressTitle,
+			} );
+
+			const planProductObject = getPlanProduct( planObject?.periodAgnosticSlug, billingPeriod );
+
+			const cartKey = await cartManagerClient.getCartKeyForSiteSlug( newSite?.site_slug as string );
+
+			// eslint-disable-next-line @typescript-eslint/no-explicit-any
+			const productsToAdd: any[] = [
+				{
+					product_slug: planProductObject?.storeSlug,
+					extra: {
+						signup_flow: flow,
+					},
+				},
+				{
+					product_slug: woothemesProduct.productSlug,
+					extra: {
+						signup_flow: flow,
+					},
+				},
+			];
+
+			if ( domain && domain.product_slug ) {
+				const registration = domainRegistration( {
+					domain: domain.domain_name,
+					productSlug: domain.product_slug as string,
+					extra: { privacy_available: domain.supports_privacy },
+				} );
+
+				productsToAdd.push( registration );
+			}
+
+			await cartManagerClient.forCartKey( cartKey ).actions.addProductsToCart( productsToAdd );
+			const redirectTo = encodeURIComponent(
+				`/setup/sensei/senseiLaunch?siteSlug=${ newSite?.site_slug }&siteId=${ newSite?.blogid }`
+			);
+
+			window.location.replace(
+				`/checkout/${ newSite?.site_slug }?signup=1&redirect_to=${ redirectTo }`
+			);
```

Comment:
I'd return just to step execution.
```suggestion
			return window.location.replace(
				`/checkout/${ newSite?.site_slug }?signup=1&redirect_to=${ redirectTo }`
			);
```